### PR TITLE
Fix unsupported Kafka version error messages in some edge cases

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Bugfix

### Description

In the past, we worked to improve the error message when a user specifies a Kafka version which is not supported. There seem to be still an edge case when a user gets very ugly error message:

```
2022-01-03 17:17:48 ERROR AbstractOperator:247 - Reconciliation #34(timer) Kafka(myproject/my-cluster): createOrUpdate failed
java.lang.IllegalArgumentException: argument "src" is null
	at com.fasterxml.jackson.databind.ObjectMapper._assertNotNull(ObjectMapper.java:4737) ~[com.fasterxml.jackson.core.jackson-databind-2.11.3.jar:2.11.3]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3504) ~[com.fasterxml.jackson.core.jackson-databind-2.11.3.jar:2.11.3]
	at io.strimzi.operator.cluster.model.KafkaConfiguration.readConfigModel(KafkaConfiguration.java:115) ~[io.strimzi.cluster-operator-0.28.0-SNAPSHOT.jar:0.28.0-SNAPSHOT]
	at io.strimzi.operator.cluster.model.KafkaConfiguration.validate(KafkaConfiguration.java:92) ~[io.strimzi.cluster-operator-0.28.0-SNAPSHOT.jar:0.28.0-SNAPSHOT]
	at io.strimzi.operator.cluster.model.KafkaCluster.validateConfiguration(KafkaCluster.java:648) ~[io.strimzi.cluster-operator-0.28.0-SNAPSHOT.jar:0.28.0-SNAPSHOT]
	at io.strimzi.operator.cluster.model.KafkaCluster.fromCrd(KafkaCluster.java:424) ~[io.strimzi.cluster-operator-0.28.0-SNAPSHOT.jar:0.28.0-SNAPSHOT]
	at io.strimzi.operator.cluster.operator.assembly.KafkaAssemblyOperator$ReconciliationState.lambda$getKafkaClusterDescription$62(KafkaAssemblyOperator.java:1667) ~[io.strimzi.cluster-operator-0.28.0-SNAPSHOT.jar:0.28.0-SNAPSHOT]
	at io.vertx.core.impl.future.Composition.onSuccess(Composition.java:38) ~[io.vertx.vertx-core-4.2.1.jar:4.2.1]
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60) ~[io.vertx.vertx-core-4.2.1.jar:4.2.1]
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211) ~[io.vertx.vertx-core-4.2.1.jar:4.2.1]
	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23) ~[io.vertx.vertx-core-4.2.1.jar:4.2.1]
	at io.vertx.core.impl.future.PromiseImpl.onSuccess(PromiseImpl.java:49) ~[io.vertx.vertx-core-4.2.1.jar:4.2.1]
	at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54) ~[io.vertx.vertx-core-4.2.1.jar:4.2.1]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) [io.netty.netty-common-4.1.71.Final.jar:4.1.71.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469) [io.netty.netty-common-4.1.71.Final.jar:4.1.71.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:503) [io.netty.netty-transport-4.1.71.Final.jar:4.1.71.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) [io.netty.netty-common-4.1.71.Final.jar:4.1.71.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [io.netty.netty-common-4.1.71.Final.jar:4.1.71.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [io.netty.netty-common-4.1.71.Final.jar:4.1.71.Final]
	at java.lang.Thread.run(Thread.java:829) [?:?]
```

This is when unsupported version is specified but at the same time a custom image is specified. The current error message is thrown from the container image lookup which is not called with custom images and does not trigger the error. Instead, the configuration validation fails because it fails to find the configuration model.

This PR does several changes to the code:
1) It adds explicit validation of the Kafka version to the `KafkaCluster` `fromCrd` method which is always called. This should avoid any possible edge cases and combinations when the container image lookup would not be called.
2) It fixes the `KafkaConfiguration` class and in case it happens and the configuration model is missing, is doesn't throw the ugly and confusing Jackson (see above) exception but instead throws a nice descriptive exception about the configuration model missing:
   ```
   2022-01-03 17:19:49 ERROR AbstractOperator:247 - Reconciliation #2(watch) Kafka(myproject/my-cluster): createOrUpdate failed
   java.lang.RuntimeException: Configuration model /kafka-2.7.1-config-model.json was not found
        at io.strimzi.operator.cluster.model.KafkaConfiguration.readConfigModel(KafkaConfiguration.java:123) ~[io.strimzi.cluster-operator-0.28.0-SNAPSHOT.jar:0.28.0-SNAPSHOT]
        at io.strimzi.operator.cluster.model.KafkaConfiguration.validate(KafkaConfiguration.java:92) ~[io.strimzi.cluster-operator-0.28.0-SNAPSHOT.jar:0.28.0-SNAPSHOT]
        at io.strimzi.operator.cluster.model.KafkaCluster.validateConfiguration(KafkaCluster.java:648) ~[io.strimzi.cluster-operator-0.28.0-SNAPSHOT.jar:0.28.0-SNAPSHOT]
        at io.strimzi.operator.cluster.model.KafkaCluster.fromCrd(KafkaCluster.java:424) ~[io.strimzi.cluster-operator-0.28.0-SNAPSHOT.jar:0.28.0-SNAPSHOT]
        at io.strimzi.operator.cluster.operator.assembly.KafkaAssemblyOperator$ReconciliationState.lambda$getKafkaClusterDescription$62(KafkaAssemblyOperator.java:1667) ~[io.strimzi.cluster-operator-0.28.0-SNAPSHOT.jar:0.28.0-SNAPSHOT]
        at io.vertx.core.impl.future.Composition.onSuccess(Composition.java:38) ~[io.vertx.vertx-core-4.2.1.jar:4.2.1]
        at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60) ~[io.vertx.vertx-core-4.2.1.jar:4.2.1]
        at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211) ~[io.vertx.vertx-core-4.2.1.jar:4.2.1]
        at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23) ~[io.vertx.vertx-core-4.2.1.jar:4.2.1]
        at io.vertx.core.impl.future.PromiseImpl.onSuccess(PromiseImpl.java:49) ~[io.vertx.vertx-core-4.2.1.jar:4.2.1]
        at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54) ~[io.vertx.vertx-core-4.2.1.jar:4.2.1]
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) [io.netty.netty-common-4.1.71.Final.jar:4.1.71.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469) [io.netty.netty-common-4.1.71.Final.jar:4.1.71.Final]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:503) [io.netty.netty-transport-4.1.71.Final.jar:4.1.71.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) [io.netty.netty-common-4.1.71.Final.jar:4.1.71.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [io.netty.netty-common-4.1.71.Final.jar:4.1.71.Final]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [io.netty.netty-common-4.1.71.Final.jar:4.1.71.Final]
        at java.lang.Thread.run(Thread.java:829) [?:?]
   ```
   In any case, this should not show up when user specifies unsupported Kafka version => only when due to some build issue the configuration model is actually missing.
3) The validation in the container image lookup is still kept in place and it might be thrown when the configuration of container images per-version is invalid.
4) It also updates the error message slightly to make it more clear that it is a Kafka version the error is talking about.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

**Review by sknot-rh — APPROVED** [review]

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaVersion.java [hunk 1]
```diff
@@ -264,7 +264,7 @@ public String kafkaMirrorMakerImage(String image, String version) {
         }
 
         InvalidResourceException asInvalidResourceException(String version, NoImageException e) {
-            return new InvalidResourceException("Version " + version + " is not supported. " +
+            return new InvalidResourceException("Kafka version " + version + " is not supported. " +
```
Comment (Frawless) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaVersion.java:267]: I think this breaks `SpecificST#testDeployUnsupportedKafka`

**Review by Frawless — APPROVED** [review]

**Pull request merged**